drivers/disk/ramdisk.c

```diff
@@ -33,6 +37,21 @@ static int disk_ram_access_status(struct disk_info *disk)
 
 static int disk_ram_access_init(struct disk_info *disk)
 {
+#if IS_ENABLED(CONFIG_DISK_RAM_EXTERNAL)
+	mm_reg_t virt_start;
+
+	if (!ramdisk_buf) {
+#if defined(CONFIG_MMU) || defined(CONFIG_PCIE)
+		device_map(&virt_start, CONFIG_DISK_RAM_START,
```

Comment:
Why can't you put `virt_start` here with explicit casting for suppressing warnings and reduce the diff?